File: e-dashboard-ui/src/ClientApp/src/views/Diagram/DiagramLib/Shapes/LineConnectShape.ts
import { DrawingShape } from './DrawingShape';
import { ILine, IPoint, IPolyLine } from './interfaces';
import { DrawingToolType } from '../Enums';
import { Point } from './Shapes';
import { ILineConnectShape } from './interfaces';

export abstract class LineConnectShape<S extends ILine | IPolyLine> extends DrawingShape<S> implements ILineConnectShape {
    public srcShapeId: string | undefined;
    public desShapeId: string | undefined;
    public srcPortName: string | undefined;
    public desPortName: string | undefined;
    public connectShapeType: string | undefined;

    constructor(name: string) {
        super(name);
        this.$type = "LineConnectShape";
        this.srcShapeId = undefined;
        this.desShapeId = undefined;
        this.srcPortName = undefined;
        this.desPortName = undefined;
        this.connectShapeType = undefined;
    }

    public draw(ctx: CanvasRenderingContext2D):void{
    }

    public move(to: IPoint): void{
    }

    public resizeToLocation(to: IPoint, handle: number): void{
    }
    public getCursorType(mousePoint: IPoint, tool: DrawingToolType): string{
        return "auto";
    }
    public getMoveOffset(mousePos: IPoint): IPoint{
        return new Point(0,0);
    }
    public toSVG(): string{
        return "";
    }
}

File: e-dashboard-api/src/Domain/Diagram/Shapes/LineConnectShape.cs


namespace EnergyDashboard.Domain.Diagram.Shapes;

public class LineConnectShape: DrawingShapeBase
{
    public LineConnectShape() : base()
    {
        this.SrcShapeId = string.Empty;
        this.SrcPortName = string.Empty;
        this.DesShapeId = string.Empty;
        this.DesPortName = string.Empty;
        this.ConnectShapeType = string.Empty;


    }
    public LineConnectShape(string name) : base(name)
    {
        this.SrcShapeId = string.Empty;
        this.SrcPortName = string.Empty;
        this.DesShapeId = string.Empty;
        this.DesPortName = string.Empty;
        this.ConnectShapeType = string.Empty;
    }

    public string SrcShapeId { get; set; }
    public string DesShapeId { get; set; }
    public string SrcPortName { get; set; }
    public string DesPortName { get; set; }
    public string ConnectShapeType { get; set; }
}
